lib/os.js

```diff
@@ -184,9 +183,11 @@ function tmpdir() {
            process.env.TMP ||
            (process.env.SystemRoot || process.env.windir) + '\\temp';
 
-    if (path.length > 1 && StringPrototypeEndsWith(path, '\\') &&
-        !StringPrototypeEndsWith(path, ':\\'))
-      return StringPrototypeSlice(path, 0, -1);
+    if (path.length > 1 && path[path.length - 1] === '\\' &&
+            !(path.length > 2 && path[path.length - 2] === ':' && path[path.length - 1] === '\\')) {
+      return path.slice(0, -1);
```

Comment:
What about primordials?

Reply:
yes I used this can you look at my last commit, thank you